src/plugins/discover/public/application/apps/main/components/sidebar/discover_field.tsx

```diff
@@ -29,6 +29,172 @@ import { IndexPatternField, IndexPattern } from '../../../../../../../data/publi
 import { getFieldTypeName } from './lib/get_field_type_name';
 import { DiscoverFieldDetailsFooter } from './discover_field_details_footer';
 
+function wrapOnDot(str?: string) {
+  // u200B is a non-width white-space character, which allows
+  // the browser to efficiently word-wrap right after the dot
+  // without us having to draw a lot of extra DOM elements, etc
+  return str ? str.replace(/\./g, '.\u200B') : '';
+}
+
+const FieldInfoIcon: React.FC = memo(() => (
```

Comment:
:information_source: I've extracted most of the inner functions returning JSX elements into proper React components within this file, so we can use `React.memo` on them.